# Identifying Sequence Motifs to Better Predict Folding  
mebipred: identifying metal-binding potential in protein sequence | Bioinformatics | Oxford Academic  
https://academic.oup.com/bioinformatics/article/38/14/3532/6594112
- metal-binding sites in proteins are usually a shell of hydrophilic residues that can be identified in the structure 
- structure-based methods are limited by the fact that a single motif could be representative of many different binding patterns and that there’s a limited number of experimentally determined protein structures available for analysis
- this is still better than sequence-based methods
- homology modeling methods perform poorly when predicting sequences modeled with structural templates
- better with a single ion instead of multiple
- computational predictions are similar to prediction of other functional characteristics of proteins (ex. mutation effects or residue importance) – uses evolutionary profiles, predicted structures, physiochem properties, and sequence descriptures as features in machine learning – has varied precision
- sequence based methods are misleading because homologs can evolve to bind different metals due to changing environmental pressures
- can predict using sequence conservation of residues near those directly interacting with ions 

Virtual Screening Strategies in Drug Discovery: A Critical Review | Bentham Science 
https://www.eurekaselect.com/article/53238 
