# An Exploration of AlphaFold
We have set out to make a fun (for us to do, and for others to see), useful (at least to us, but hopefully to you and others), relevant (to the class, to us, to humanity), and technically interesting project exploring an innovative program called AlphaFold. 

How proteins fold has been a question that biologists, chemists, and statisticians have sought to answer for years. The biochemistry of proteins have been well-studied since the 1800s and since then, advances in biological and chemical techniques and knowledge coupled with better control of modeling and machine learning have resulted in protein motif/domain discovery algorithms and protein structure predictors. 

AlphaFold is an incredibly useful tool that uses multiple-sequence alignment to predict protein structure. The most recent version of Alphafold (Alphafold2) uses coupled sub-networks based on pattern recognition and relationships between amino acids in a concept-dependent manner as derived from training data. Alphafold2, while better at predicting protein structure than Alphafold1, is not good at predicting the structure of proteins that are intrinsically disordered (fold differently in the presence of different protein subunits), multi-unit heteromeric proteins, or proteins that interact with organic molecules, metals, or co-factors in order to assume the appropriate conformation. While this exploration focuses on Alphafold and motif identification techniques that could help Alphafold integrate non-protein and inorganic compounds into its predicted structure, there are many other structural predictive algorithms that exist for many different types of proteins. 

Understanding the 3D structure of proteins is incredibly useful, as proteins form the basis of most drug and therapeutic treatment targets, but deriving this structures experimentally can be cost-prohibitive. Predictive modeling helps decrease the cost of this critical component of protein analysis and dynamics, and with further development, modeling can become more accurate and useful for pharmacology and de novo protein synthesis. 
  
Project Members: Asheeta Bothra (<ab3wjt@virginia.edu>), Emily Kao (<eck3pxj@virginia.edu>), Meesha Vullikanti (<rv6cun@virginia.edu>) <br>  
1. [Proteins and the Protein-Folding Problem](https://eckao.github.io/compbio-alphafold-project/proteinstructures)
2. [Identifying Sequence Motifs to Better Predict Folding](https://eckao.github.io/compbio-alphafold-project/bindingmotifs)
3. [What is AlphaFold?](https://eckao.github.io/compbio-alphafold-project/alphafold)
4. [Successes and Limitations of AlphaFold](https://eckao.github.io/compbio-alphafold-project/shortcomings).  
5. [References](https://eckao.github.io/compbio-alphafold-project/references)  

![AlphaFold Image](./docs/assets/alphafold_image.jpeg "AlphaFold Image")  
Image source: <https://scitechdaily.com/deepmind-ai-powers-major-scientific-breakthrough-alphafold-generates-3d-view-of-the-protein-universe/>
